src/coreclr/jit/valuenum.cpp

```diff
@@ -8444,14 +8444,16 @@ void Compiler::fgValueNumberSimd(GenTree* tree)
     assert(tree->OperGet() == GT_SIMD);
     GenTreeSIMD* simdNode = tree->AsSIMD();
     assert(simdNode != nullptr);
+
+    VNFunc       simdFunc = GetVNFuncForNode(tree);
     ValueNumPair excSetPair;
     ValueNumPair normalPair;
 
     // There are some SIMD operations that have zero args, i.e.  NI_Vector128_Zero
     if (tree->AsOp()->gtOp1 == nullptr)
     {
         excSetPair = ValueNumStore::VNPForEmptyExcSet();
-        normalPair = vnStore->VNPairForFunc(tree->TypeGet(), GetVNFuncForNode(tree));
+        normalPair = vnStore->VNPairForFunc(tree->TypeGet(), simdFunc);
```

Comment:
Do we need to do anything special for operations like `StoreFence`/`LoadFence`/`MemoryFence` which, while not load/stores likely shouldn't be CSE'd?

Reply:
Ah, this is handling `GT_SIMD` not `GT_HWINTRINSIC` so the comment would apply elsewhere.

Reply:
As this change might need to be backported to 5.0, I am limiting it to the minimal to fix the issue.

Currently all HWIntrinsic Loads are not CSE candidates,  I have an additional change that allows them to be CSE'd.
Only the non special ones can be made CSE candidates.

The special operations like PrefetchX, LoadFence,  etc.  will never be CSE candidates

(See OptCSE.cpp line 3677)